src/relations.js

```diff
@@ -0,0 +1,48 @@
+import { deserialize } from "./serialize.js";
+
+/* global window, document, Zotero, Lidia */
+
+/**
+ * Get an array of all annotations in a given Zotero library
+ * @param {number} libraryID - the requested library ID
+ * @return {Array.<Object>} - a list of all annotations as JavaScript objects
+ */
+export async function getAllLidiaAnnotations(libraryID) {
+    if (!libraryID) {
+        libraryID = Zotero.Libraries.userLibraryID;
+    }
+    /* Find all items in the requested library; get their attached PDFs and
+     * in turn all their annotations. Check for each annotation if it is a
+     * LIDIA annotation and return a list of all of these
+     */
+    const s = new Zotero.Search();
+    s.libraryID = libraryID;
+    s.addCondition('recursive', 'true');
```

Comment:
I wonder if we could use `s.addCondition('itemType', 'is', 'annotation');` to get all annotations (not tested), and then get the `parentItem` to get the publication title and key per annotation. If it's possible, it should be faster on large libraries and require less nesting.

Reply:
This does not work -- apparently this does not give back the annotations but a list of the PDF items that belong to them (I have not looked into that in detail).